### ISSUE TITLE
ZipInputStream.GetNextEntry hangs permanently

### ISSUE THREAD

**Issue by Metalnem** [icsharpcode/SharpZipLib#300]
**ZipInputStream.GetNextEntry** hangs permanently when attempting to extract the [attached archive]. I'm using the latest SharpZipLib NuGet package (version 1.1.0). You can run the following code to reproduce it (the **path** variable should contain the path to the attached file):

```csharp
using (var file = File.OpenRead(path))
using (var zip = new ZipInputStream(file))
{
  while (zip.GetNextEntry() != null) { }
}
```

Found via [SharpFuzz].

[attached archive]: https://github.com/icsharpcode/SharpZipLib/files/2708827/Test.zip
[SharpFuzz]: https://github.com/Metalnem/sharpfuzz

**Comment by Numpsy** [comment]
Looks to be getting stuck @ https://github.com/icsharpcode/SharpZipLib/blob/master/src/ICSharpCode.SharpZipLib/Zip/ZipInputStream.cs#L372, where Read() constantly returns more data.

Possibly related to https://github.com/icsharpcode/SharpZipLib/issues/128 / https://github.com/icsharpcode/SharpZipLib/issues/88 / https://github.com/icsharpcode/SharpZipLib/issues/19 ?

**Comment by piksel** [comment]
Fixed by #316.
